tests/acceptance/core/service/assert/pod.go

```diff
@@ -0,0 +1,76 @@
+package assert
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/k3s-io/k3s/tests/acceptance/shared/util"
+	. "github.com/onsi/gomega"
+)
+
+// PodAssertFunc is a function type used to create pod assertions
+type PodAssertFunc func(g Gomega, pod util.Pod)
+
+// PodAssertRestart custom assertion func that asserts that pods are not restarting with no reason
+// controller, scheduler, helm-install pods can be restarted occasionally when cluster started if only once
+func PodAssertRestart() PodAssertFunc {
+	return func(g Gomega, pod util.Pod) {
+		if strings.Contains(pod.NameSpace, "kube-system") &&
+			strings.Contains(pod.Name, "controller") &&
+			strings.Contains(pod.Name, "scheduler") {
+			g.Expect(pod.Restarts).Should(SatisfyAny(Equal("0"),
+				Equal("1")),
+				"could be restarted occasionally when cluster started", pod.Name)
+		}
+	}
+}
+
+// PodAssertStatus custom assertion that asserts that pods status is completed or in some cases
+// apply pods can have error status
+func PodAssertStatus() PodAssertFunc {
+	return func(g Gomega, pod util.Pod) {
+		if strings.Contains(pod.Name, "helm-install") {
+			g.Expect(pod.Status).Should(Equal(util.CompletedAssert), pod.Name)
```

Comment:
using constants for keywords like "Complated" or "Running" in pod status makes it complex and less readable.